### PR TITLE
Added presence AnimeUltima

### PR DESCRIPTION
**Pull request by VocalZero**
Implemented the presence request from coolads934. #3872

## Proof of work
https://www1.animeultima.to/
Screenshot 2021-06-09 205959

https://www1.animeultima.to/search?search=pokemon
Screenshot 2021-06-09 210103

https://www1.animeultima.to/a/pokemon-2019_281673
Screenshot 2021-06-09 210126

https://www1.animeultima.to/a/kyuukyoku-shinka-shita-full-dive-rpg-ga-genjitsu-yori-mo-kusoge-dattara_290916/episode-1_858541-dub
Screenshot 2021-06-09 210154

### CODE REVIEW

File: websites/A/AnimeUltima/presence.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+function clearPresenceData() {
+  delete presenceData.smallImageKey;
+  delete presenceData.state;
+  delete presenceData.details;
+  delete presenceData.startTimestamp;
+}
+
+presence.on("UpdateData", () => {
+  clearPresenceData();
+  presenceData.startTimestamp = browsingTimestamp;
```
Comment (ririxidev) [inline websites/A/AnimeUltima/presence.ts:19]: delete it please... and use presenceData inside event and add `startTimestamp` to presenceData object

File: websites/A/AnimeUltima/presence.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    presenceData.smallImageKey = "playing";
+    presenceData.buttons = [ { label: "Watch Episode", url: document.location.toString() } ];
+    presence.setActivity(presenceData, true);
+  } else if(document.location.pathname.includes("/a/")) {
+    const [title] = document.getElementsByClassName("title is-marginless is-paddingless");
+    presenceData.details = "Searching...";
+    presenceData.state = title.textContent;
+    presenceData.smallImageKey = "browsing";
+    presenceData.buttons = [ { label: "Watch Anime", url: document.location.toString() } ];
+    presence.setActivity(presenceData, true);
```
Comment (ririxidev) [inline websites/A/AnimeUltima/presence.ts:34]:
```suggestion
    presence.setActivity(presenceData);
```

File: websites/A/AnimeUltima/presence.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    presenceData.state = currentEpisode ?? "Episode not found...";
+    presenceData.smallImageKey = "playing";
+    presenceData.buttons = [ { label: "Watch Episode", url: document.location.toString() } ];
+    presence.setActivity(presenceData, true);
+  } else if(document.location.pathname.includes("/a/")) {
+    const [title] = document.getElementsByClassName("title is-marginless is-paddingless");
+    presenceData.details = "Searching...";
+    presenceData.state = title.textContent;
+    presenceData.smallImageKey = "browsing";
+    presenceData.buttons = [ { label: "Watch Anime", url: document.location.toString() } ];
```
Comment (ririxidev) [inline websites/A/AnimeUltima/presence.ts:33]:
```suggestion
    presenceData.buttons = [ { label: "View Anime", url: document.location.toString() } ];
```

File: websites/A/AnimeUltima/presence.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const presence = new Presence({
+  clientId: "852239305983262761"
+}),
+browsingTimestamp = Date.now(),
```
Comment (ririxidev) [inline websites/A/AnimeUltima/presence.ts:4]:
```suggestion
browsingTimestamp = ~~(Date.now() / 1000),
```

**Review by rxri — CHANGES_REQUESTED** [review]
also isn't there more subpages to support :/
and support player too :////

**Comment by VocalZero** [comment]
I could add a few more subpages. That's not a big problem but with the support of the player I don't get further at the moment. When I try to view the page, the player stops and is paused by Chrome Debug. I have tried a few methods but have not been able to access the video object. Would anyone have an idea how I could implement this?

**Comment by ririxidev** [comment]
https://greasyfork.org/en/scripts/32015-anti-anti-debugger/code

**Comment by VocalZero** [comment]
I'm now trying to get to the stuff in the iFrame with "document.querySelector("iframe").contentWindow.document" but I'm getting a cross-origin frame error. Somehow I don't like the page and the page doesn't like me :/

**Comment by VocalZero** [comment]
I have now tried again but without success. The Cross-Origin Frame error blocks me from accessing the iFrame. Does anyone know a fix? Otherwise I leave the player away from the presence.

**Comment by ririxidev** [comment]
```document.querySelector("#app > section.section.dark-container.is-paddingless > div > div.player-container.player-container-16x9 > iframe")```
and no draft prs.

**Comment by VocalZero** [comment]
Sorry about the pr draft. Again to the iFrame I'm already so far. I try from the iFrame to get the video to take some data from it but somehow I can't get any further because of the error I mentioned earlier.

```typescript
const iframe = document.querySelector("#app > section.section.dark-container.is-paddingless > div > div.player container.player-container-16x9 > iframe"),
   video = iframe.contentWindow.document.querySelector("video");
console.log(video); // returns null because of error Blocked a frame with origin "https://www1.animeultima.to" from accessing a cross-origin frame.
```

**Review by rxri — APPROVED** [review]
okay then for now

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**